File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        
        int n = sizes.length;
        
        int max[] = new int[n];
        int min[] = new int[n];
        
        for (int i = 0; i < n; i++) {
            max[i] = Math.max(sizes[i][0], sizes[i][1]);
            min[i] = Math.min(sizes[i][0], sizes[i][1]);
        }
        
        Arrays.sort(max);
        Arrays.sort(min);
        
        return max[n-1] * min[n-1];
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int n = sizes.size();
    int answer = 0;
    vector<int> maxSize(n);
    vector<int> minSize(n);
    
    for (int i = 0; i < sizes.size(); i++) {
        maxSize[i] = max(sizes[i][0], sizes[i][1]);
        minSize[i] = min(sizes[i][0], sizes[i][1]);
    }
    
    answer = (*max_element(maxSize.begin(), maxSize.end()) * *max_element(minSize.begin(), minSize.end()));        
    
    return answer;
}